db_test.go

```diff
@@ -51,6 +51,11 @@ func openTestDB(t testing.TB, opts *Options) (db *DB, close func()) {
 // query runs a matcher query against the querier and fully expands its data.
 func query(t testing.TB, q Querier, matchers ...labels.Matcher) map[string][]sample {
 	ss, err := q.Select(matchers...)
+	defer func() {
```

Comment:
I see the querier is closed wherever `query()` is called, so I think this change is not required. Also, I would not expect querier to be closed inside a function where I pass the querier.